filebeat/beat/spooler.go

```diff
@@ -112,6 +113,12 @@ func (s *Spooler) Stop() {
 
 // flush flushes all event and sends them to the publisher
 func (s *Spooler) flush() {
+
+	// Prevents race in flush method
+	var mutex = &sync.Mutex{}
```

Comment:
If `flush()` creates a new Mutex on each invocation it can't really protect against a data race. The mutex should be part of the Spooler.

I'm not sure what piece of data is causing the race. The line numbers in the trace didn't help much because it says line 140 but the file doesn't even have 140 lines (weird?). If the data race is caused by multiple goroutines reading and writing to/from `s.nextFlushTime` then the mutex will need to be locked around the read inside the `Run()` method.